spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala

```diff
@@ -213,22 +213,50 @@ abstract class AbstractModule[A <: Activity: ClassTag, B <: Activity: ClassTag,
   }
 
   /**
-   * freeze the layer, its parameters(weight/bias, if exists)
-   * are not changed in training process
+   * freeze the module,
+   * i.e. their parameters(weight/bias, if exists) are not changed in training process
+   * if names is not null,
+   * set an array of layers that match the given ```names``` to be "freezed",
+   *
+   * @param names an array of layer names
+   * @return current graph model
    */
-  def freeze(): this.type = {
-    setScaleW(0)
-    setScaleB(0)
+  def freeze(names: Array[String] = null): this.type = {
```

Comment:
ugly. Don't use null for some meaning in a public API! use var length String instead